kraken-std/src/kraken/std/python/settings.py

```diff
@@ -39,6 +39,7 @@ class _PackageIndex(PackageIndex):
     build_system: PythonBuildSystem | None = None
     source_directory: Path = Path("src")
     tests_directory: Path | None = None
+    lint_enforced_directories: list[Path] | None = None
```

Comment:
As it is a sequence, we can make it non nullable and default to an empty sequence: it will make code in the tasks a bit more simple because then we can just always iterate through `lint_enforced_directories` without having to check for `None` first.

```suggestion
    lint_enforced_directories: list[Path] = dataclasses.field(default_factory=list)
```

By doing that we lose the ability to distinguish between "this value has been provided but it is empty" and "this value has not been provided", but I think this is acceptable (and arguably: desirable :)).

Reply:
I like this, no reason we can't be more specific with the `PythonSettings` object compared to with `python_settings`. Thanks for the suggestion, I have made the change.